app/src/js/components/Input/edit.js

```diff
@@ -0,0 +1,165 @@
+'use strict';
+import React from 'react';
+import AceEditor from 'react-ace';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { withRouter, Link } from 'react-router-dom';
+import {
+  getInput,
+  createInput,
+  updateInput
+} from '../../actions';
+import config from '../../config';
+import Loading from '../LoadingIndicator/loading-indicator';
+import Breadcrumbs from '../Breadcrumbs/Breadcrumbs';
+import ErrorReport from '../Errors/report';
+import { formPrivilegesCU } from '../../utils/privileges';
+
+class Questions extends React.Component {
+  constructor () {
+    super();
+    this.state = { view: 'json', data: '', section_data: '' };
+    this.refName = React.createRef();
+    this.sectionRefName = React.createRef();
+    this.renderQuestionJson = this.renderQuestionJson.bind(this);
+    this.renderJson = this.renderJson.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  async componentDidMount () {
+    const { inputId, controlId } = this.props.match.params;
+    await this.props.dispatch(getInput(inputId, controlId))
```

Comment:
Yes, it is question_id. As I said, the unique key is a combination of control_id and question_id.

Do you want me to rename it to questionId instead of inputId?